references.go

```diff
@@ -82,41 +80,46 @@ func walkGraph(result *[]*object.Commit, seen *map[plumbing.Hash]struct{}, curre
 	if err != nil {
 		return err
 	}
-	switch len(parents) {
-	// if the path is not found in any of its parents, the path was
-	// created by this commit; we must add it to the revisions list and
-	// stop searching. This includes the case when current is the
-	// initial commit.
-	case 0:
+	if len(parents) == 0 {
+		// if the path is not found in any of its parents, the path was
+		// created by this commit; we must add it to the revisions list and
+		// stop searching. This includes the case when current is the
+		// initial commit.
 		*result = append(*result, current)
 		return nil
-	case 1: // only one parent contains the path
-		// if the file contents has change, add the current commit
-		different, err := differentContents(path, current, parents)
+	}
+	// or more than one parent contains the path
+	skip, err := derivedFromAnyParent(path, current, parents)
```

Comment:
I still think this function should have a better name, or at least an explanation. Just looking at this line, I have no idea what is going on, `skip = derivedFromAnyParent`, huh?

Reply:
Ok, I think I figured it out, maybe something along the lines of `same, err := sameContentAsAnyParent`?

Reply:
It could be a merged commit so the file comes from one parent directly, so I think `derived` marks that the file is from a parent not by coincidence.